File: ts/audit/ledger.ts
import crypto from "crypto";

export interface AuditEntry {
  timestamp: string;
  sessionId: string;
  decision: string;
  prevHash: string;
  hash: string;
}

export function createAuditEntry(
  sessionId: string,
  decision: string,
  prevHash: string
): AuditEntry {

  const timestamp = new Date().toISOString();
  const payload = `${timestamp}|${sessionId}|${decision}|${prevHash}`;

  const hash = crypto
    .createHash("sha256")
    .update(payload)
    .digest("hex");

  return { timestamp, sessionId, decision, prevHash, hash };
}

File: py/audit/ledger.py
import hashlib
from datetime import datetime

def create_audit_entry(session_id, decision, prev_hash):
    ts = datetime.utcnow().isoformat()
    payload = f"{ts}|{session_id}|{decision}|{prev_hash}"
    h = hashlib.sha256(payload.encode()).hexdigest()

    return {
        "timestamp": ts,
        "sessionId": session_id,
        "decision": decision,
        "prevHash": prev_hash,
        "hash": h
    }